x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/index.tsx

```diff
@@ -124,9 +129,17 @@ export const termsOperation: OperationDefinition<TermsIndexPatternColumn, 'field
     if (field && !isScriptedField(field)) {
       secondaryFields.add(field.name);
     }
-    return {
-      secondaryFields: [...secondaryFields].filter((f) => targetColumn.sourceField !== f),
+    const secondaryFieldsList: string[] = [...secondaryFields].filter(
```

Comment:
why do you copy the array here to filter it?

Reply:
Because it's a `Set` and I'm making it an array out of it. But actually this note made me think to rewrite this in another way that removes one cycle.